engines/agos/metaengine.cpp

```diff
@@ -207,6 +213,187 @@ SaveStateList AgosMetaEngine::listSaves(const char *target) const {
 
 int AgosMetaEngine::getMaximumSaveSlot() const { return 999; }
 
+Common::KeymapArray AgosMetaEngine::initKeymaps(const char *target) const {
+	using namespace Common;
+	using namespace AGOS;
+
+	Common::String gameId = ConfMan.get("gameid", target);
+	
+	Keymap *engineKeyMap = new Keymap(Keymap::kKeymapTypeGame, "agos-main", _("AGOS main"));
+	Keymap *gameKeyMap = new Keymap(Keymap::kKeymapTypeGame, "game-shortcuts", _("Game Keymappings"));
+	Keymap *gameKeyMap2 = new Keymap(Keymap::kKeymapTypeGame, "game-shortcuts-2", _("Game Keymappings 2"));
```

Comment:
I'd recommend splitting the keymaps this way:
* The main keymap with mouse button and cancel actions, which is always active
* The game keymap with the in-game shortcuts, which is enabled by default, but disabled when text input is active
* The yes/no keymap, which is disabled by default, but enabled during `AGOSEngine_Simon1::os1_pauseGame()`. This way, it should be possible to map those actions to joystick buttons that are otherwise used elsewhere.